homeassistant/util/json.py

```diff
@@ -64,10 +77,20 @@ def save_json(
 
     Returns True on success.
     """
-    dump: Callable[[Any], Any] = json.dumps
+    dump: Callable[[Any], Any]
     try:
         if encoder:
-            json_data = json.dumps(data, indent=2, cls=encoder)
+            # For backwards compatibility, if they pass in the
+            # default json encoder we use _orjson_default_encoder
+            # which is the orjson equivalent to the default encoder.
+            if encoder == DefaultHASSJSONEncoder:
```

Comment:
I think that we just need to verify that it's the exact same reference

```suggestion
            if encoder is DefaultHASSJSONEncoder:
```